plugin/trino-hive/src/main/java/io/trino/plugin/hive/security/LegacyAccessControl.java

```diff
@@ -399,9 +398,13 @@ public void checkCanExecuteTableProcedure(ConnectorSecurityContext context, Sche
     @Override
     public void checkCanExecuteFunction(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName function)
     {
-        if (functionKind == TABLE) {
-            denyExecuteFunction(function.toString());
+        switch (functionKind) {
+            case SCALAR, AGGREGATE, WINDOW:
+                return;
+            case TABLE:
+                denyExecuteFunction(function.toString());
```

Comment:
>  // TODO implement

I guess if it's `LegacyAccessControl` so it should not be implemented here?

Reply:
agree